app/src/main/java/org/mozilla/fenix/home/collections/TabInCollection.kt

```diff
@@ -0,0 +1,300 @@
+        }
+
+        Card(
+            modifier = clippingModifier
+                .fillMaxWidth()
+                .defaultMinSize(minHeight = 56.dp),
+            shape = if (isLastInCollection) bottomTabShape else middleTabShape,
+            backgroundColor = FirefoxTheme.colors.layer2,
+            elevation = 5.dp,
+        ) {
+            FaviconListItem(
+                label = tab.title,
+                description = shortenUrl(tab.url),
+                onClick = { onClick() },
+                url = tab.url,
+                iconPainter = painterResource(drawable.ic_close),
+                iconDescription = stringResource(string.remove_tab_from_collection),
+                onIconClick = { onRemove(false) },
+            )
+        }
+    }
+}
+
+/**
+ * Composable used to display the background of a [Tab] shown in collections that is being swiped left or right.
+ *
+ * @param dismissDirection [DismissDirection] of the tab being swiped depending on which this composable
+ * will also indicate the swipe direction by placing a warning icon at the start of the swipe gesture.
+ * If `null` the warning icon will be shown at both ends.
+ * @param isLastInCollection Whether the tab is to be shown between others or as the last one in collection.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun DismissedTabBackground(
+    dismissDirection: DismissDirection?,
+    isLastInCollection: Boolean,
+) {
+    Card(
+        modifier = Modifier.fillMaxSize(),
+        backgroundColor = FirefoxTheme.colors.layer3,
+        shape = if (isLastInCollection) bottomTabShape else middleTabShape,
+        elevation = 0.dp,
+    ) {
+        Row(
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            Icon(
+                painter = painterResource(drawable.ic_delete),
+                contentDescription = null,
+                modifier = Modifier
+                    .padding(horizontal = 32.dp)
+                    // Only show the delete icon for where the swipe starts.
+                    .alpha(
+                        if (dismissDirection == StartToEnd) 1f else 0f
+                    ),
+                tint = FirefoxTheme.colors.iconWarning,
+            )
+
+            Icon(
+                painter = painterResource(drawable.ic_delete),
+                contentDescription = null,
+                modifier = Modifier
+                    .padding(horizontal = 32.dp)
+                    // Only show the delete icon for where the swipe starts.
+                    .alpha(
+                        if (dismissDirection == EndToStart) 1f else 0f
+                    ),
+                tint = FirefoxTheme.colors.iconWarning,
+            )
+        }
+    }
+}
+
+/**
+ * Clips the Composable this applies to such that it cannot draw content / shadows outside it's top bound.
+ */
+private fun Modifier.clipTop() = this.then(
```

Comment:
do we foresee this being used elsewhere? If so, would it make sense to have this live in the `Modifier` extension file?

Reply:
We probably don't want to keep this around longer than we have to. The plan is to join all the collection composables under a single Card. See https://github.com/mozilla-mobile/fenix/pull/24501#discussion_r869410027

Reply:
Thank you!
There are a number of bugs with the current homescreen collections (also filed recently [some](https://github.com/mozilla-mobile/fenix/issues?q=is%3Aopen+is%3Aissue+author%3AMugurell+label%3AFeature%3ACollections))  I'd say mostly stemming from the cumbersome data flow of this feature. It would need to be migrated to our MVI store and then one composable observing said store would have an easier time of syncing everything.
This patch only ment to do a 1-1 migration from the old View based system to Compose. Same bugs should still exist and work exactly as before 💪.
The above modifier was needed for a new situation in which it seems like two separate composables will show a bit of a shadow in between. May be a bug in Compose but it should also be resolved by having everything in just on ViewHolder. It does seems like a bit of a smell and I'd agree we don't want to keep it around any longer than is expressly needed.